Jenkinsfile

```diff
@@ -0,0 +1,296 @@
+def wrapperConfigurations = [
+  Debug: 'dbg',
+  Release: ''
+]
+def configuration = 'Debug'
+
+def nuget = '/usr/local/bin/nuget'
+def xbuild = '/usr/local/bin/xbuild'
+def mono = '/usr/local/bin/mono'
+
+def version
+def versionString
+
+stage('Checkout') {
+  node {
+    checkout([
+        $class: 'GitSCM',
+        branches: scm.branches,
+        gitTool: 'native git',
+        extensions: scm.extensions + [
+          [$class: 'CleanCheckout'],
+          [$class: 'SubmoduleOption', recursiveSubmodules: true]
+        ],
+        userRemoteConfigs: scm.userRemoteConfigs
+      ])
+
+      version = readAssemblyVersion()
+      versionString = "${version.major}.${version.minor}.${version.patch}"
+
+      stash includes: '**', name: 'dotnet-source'
+  }
+}
+
+def getArchive() {
+    sh 'rm -rf *'
+    unstash 'dotnet-source'
+}
+
+stage('RealmWeaver') {
+  node('xamarin-mac') {
+    getArchive()
+    def workspace = pwd()
+    sh "${nuget} restore Realm.sln"
+
+    dir('Weaver/WeaverTests/RealmWeaver.Tests') {
+      sh "${xbuild} RealmWeaver.Tests.csproj /p:Configuration=${configuration}"
+      sh "${mono} \"${workspace}\"/packages/NUnit.ConsoleRunner.*/tools/nunit3-console.exe RealmWeaver.Tests.csproj --result=TestResult.xml\\;format=nunit2 --config=${configuration} --inprocess"
+      publishTests 'TestResult.xml'
```

Comment:
We've been talking about making the tests output JUnit-compatible reports instead :/

I'd rather do that than complicate the build script, but thanks for pointing this out - we had been wondering why some builds were marked as successful on GitHub even though they were failing.